src/pyodide/internal/workers.py

```diff
@@ -682,6 +719,138 @@ async def text(self) -> str:
         return await self.js_object.text()
 
 
+def serialize_to_python(obj: "JsProxy"):
+    """
+    Converts JS objects like Response, Request, Blob, etc. to equivalent Python objects defined in
+    this module and also other JS objects like Map, Set, etc. to equivalent Python stdlib objects.
+
+    This method is used for Workers RPC in Python to convert JavaScript objects to Python. As such
+    it does not support serializing all JS object types.
+    """
+
+    def convert(obj, convert, cache):
+        if not hasattr(obj, "constructor"):
+            # Assume that the object doesn't need conversion as it's not a JS object.
+            return obj
+
+        if obj.constructor.name == "Response":
+            return Response(obj)
+        elif obj.constructor.name == "FormData":
+            return FormData(obj)
+        elif obj.constructor.name == "Blob":
+            return Blob(obj)
+        elif obj.constructor.name == "File":
+            return File(obj)
+        elif obj.constructor.name == "Request":
+            return Request(obj)
+        elif obj.constructor.name == "Date":
+            # TODO: Pyodide should gain support for this, we should upstream this.
+            return datetime.datetime.fromtimestamp(obj.getTime() / 1000)
+        elif obj.constructor.name == "Error":
+            return Exception(obj.toString())
+        elif obj.constructor.name == "Number":
+            return obj.valueOf()
```

Comment:
I don't think this is reachable? Or necessary? Numbers already get converted. Also, even in JS if x is a number, `x.valueOf()` just returns `x`, so I don't think it would do anything as JavaScript code either.

Reply:
This is for `new Number("123123")` which doesn't get converted, or at least I had to add this to make the test that sends this via RPC

Reply:
I don't believe this. If it's necessary, please make sure there is a test case that breaks when it is removed.

Reply:
It is definitely necessary, the test suite does test this path and it does fail if I remove it.

Also I just tried it in Pyodide REPL:

```
>>> x = js.Number.new("1234")
>>> isinstance(x, JsProxy)
True
>>> y = x.to_py()
>>> isinstance(y, JsProxy)
True
>>> y == 1234
False
```

Reply:
ohhhh it's so weird that `new Number("10")` and `Number("10")` both work and do really different things... I tried `js.Number("1234")` and that works the way I expected.